packages/edit-navigation/src/components/sidebar/delete-menu.js

```diff
@@ -2,31 +2,45 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, PanelBody } from '@wordpress/components';
+import {
+	Button,
+	PanelBody,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
 
 export default function DeleteMenu( { onDeleteMenu, isMenuBeingDeleted } ) {
+	const [ showConfirmDialog, setShowConfirmDialog ] = useState( false );
+
+	const handleConfirm = () => {
+		setShowConfirmDialog( false );
+		onDeleteMenu();
```

Comment:
Maybe we should `await` for `onDeleteMenu()` to finish? (I think it returns a promise?)

Bonus point if we can also catch errors here and display the message somewhere, but that's probably best for a different PR.

Another micro-optimization we can make is to show a loading state when the menu is still being deleted. Something like disabling the confirm button and showing a loading icon on the button. But again, probably best for a follow-up PR (I don't think it's possible now with the current API in `__experimentalConfirmDialog` anyway).

These are all just nitpicking though, I think we can merge this and work on those later.

Reply:
> Maybe we should `await` for `onDeleteMenu()` to finish? (I think it returns a promise?)`

Hard to say for sure without any docs in this component.

I did a quick search in the codebase, and it looks like the only usage is in [`packages/edit-navigation/src/components/layout/index.js`](https://github.com/WordPress/gutenberg/blob/11294e370b29a237080b946e405c61ed75fa9509/packages/edit-navigation/src/components/layout/index.js#L195). In that case, the `deleteMenu` function comes from the [`useNavigationEditor()` hook](https://github.com/WordPress/gutenberg/blob/ca4cadf63e28998e463615eab269b5b56cc26bd8/packages/edit-navigation/src/hooks/use-navigation-editor.js#L77-L97), and it is indeed an `async` function.

So yeah, we could technically

```js
await onDeleteMenu();
setShowConfirmDialog( false );
```

> Bonus point if we can also catch errors here and display the message somewhere, but that's probably best for a different PR.

Given the [delete function definition](https://github.com/WordPress/gutenberg/blob/ca4cadf63e28998e463615eab269b5b56cc26bd8/packages/edit-navigation/src/hooks/use-navigation-editor.js#L77-L97), it looks like errors are handled via the `createErrorNotice` function, which seems to be connected to the notices story — I'm not familiar with what's the best "gutenberg" way to handle that?

> Another micro-optimization we can make is to show a loading state when the menu is still being deleted. Something like disabling the confirm button and showing a loading icon on the button. But again, probably best for a follow-up PR (I don't think it's possible now with the current API in __experimentalConfirmDialog anyway).

We should have access to the loading state via the `isMenuBeingDeleted` prop passed to this component, but I don't think that the `ConfirmDialog` supports a way to disable its confirm button from outside the component (@fullofcaffeine can you confirm?)

Reply:
c.c. @adamziel as this is a good example of lacking a clear and simple way to do loading state + error handling with the current API 😆 . (#38135)

Reply:
> supports a way to disable its confirm button from outside the component (@fullofcaffeine can you confirm?)
You're right, that's not supported at the moment. I think it's a great improvement idea. What would that API look like do you think? A simple `isLoading` boolean prop that would disable the button and show a loading spinner when `true`?

Reply:
> A simple `isLoading` boolean prop that would disable the button and show a loading spinner when `true`?

Yeah, I think so. Since it should be a common feature, we can allow a simple prop like `isLoading` or `disabled` to do the job.

Reply:
>  A simple `isLoading` boolean prop that would disable the button and show a loading spinner when true?

I think I like the sound of `isDisabled` more as it is closer to what we want to achieve — disable interactions with the modal. Let's also add a `is` in front of the prop to make it clear that it's a boolean flag.

We also have to disable pressing the ESC key and clicking on the background, and we'll have to make this feature work both in controlled and uncontrolled mode. We can discuss further these details and refine the UX separately in the PR implementing the feature.